core/configuredivecomputerthreads.cpp

```diff
@@ -360,16 +361,36 @@ static dc_status_t write_suunto_vyper_settings(dc_device_t *device, DeviceDetail
 	EMIT_PROGRESS();
 	return rc;
 }
+#undef EMIT_PROGRESS
+
+#define EMIT_PROGRESS()	do { \
+		progress.current++; \
+		progress_cb(device_data.device, DC_EVENT_PROGRESS, &progress, userdata); \
+	} while (0)
+
+static dc_status_t ostc3_sync_time(device_data_t &data)
```

Comment:
The ostc3_sync_time name is a bit confusing. It's neither ostc or ostc3 specific. I'd suggest just something like dc_sync_time or libdivecomputer_sync_time .

It's generic and can be used on any dc that supports dc_device_timesync, which is quite a few more than just OSTC's.

Reply:
> The ostc3_sync_time name is a bit confusing. It's neither ostc or ostc3 specific. I'd suggest just something like dc_sync_time or libdivecomputer_sync_time .
> 
> It's generic and can be used on any dc that supports dc_device_timesync, which is quite a few more than just OSTC's.

In the context of the dive computer configuration, this is actually specific to the OSTC 3 (and the OSTC 4/5, which uses the 'OSTC 3' protocol - this becomes evident as even the OSTC uses different code to generate the correct time format in `write_ostc_settings()`.

Reply:
No, its not. The OSTC uses different code because it never got ported to the new libdivecomputer api's when those were introduced. The ostc3 code was ported back in 128fc39274791525113bee5ac7882352beb8cb70 but the ostc code weren't.

The timezone bits there is just a red herring introduced in 6554e4f21e4deaa6b140ff09e07f629474257308 as a equality check , which probably produced a warning which got fixed in abc6c7d2b3ff630a0c7cbc5e1aeda74919e7a078.
The backing code in hw_ostc_device_timesync doesn't care about any timezone.

My old OSTC 2N screen died a couple of years back, so I can't really test this any longer.

Reply:
We could even call sync_divecomputer_time from core/libdivecomputer.cpp here, rather than having it's own copy of a equivalent function.

Reply:
Thanks, I wasn't aware of the context.